### ISSUE TITLE
Arch = foo_bar in .plzconfig not equivalent to --arch=foo_bar

### ISSUE THREAD

**Issue by subruber** [thought-machine/please#1494]
I have a repository where I have code that ~always has to be cross-compiled; so, to avoid having to specify --arch=foo_bar every time, I tried to put it in `.plzconfig` as:

```
[Build]
Arch = foo_bar
```
In addition, I have a `.plzconfig_foo_bar `at the root of the repository with the configuration for that architecture (CppTool, etc.).

The issue is that, when Arch is specified in .plzconfig, the `.plzconfig_foo_bar` file is not read. This happens with both version 15.12.0 and with the code from the GitHub repo from a couple of days ago.

This is the output when .plzconfig does not contain Arch:

```
$ please build --arch=foo_bar -v=4
23:03:49.047   DEBUG: Found repo root at /home/user/src/test_repo
23:03:49.081   DEBUG: Attempting to read config from /etc/please/plzconfig...
23:03:49.081   DEBUG: Attempting to read config from /home/user/.config/please/plzconfig...
23:03:49.081   DEBUG: Attempting to read config from /home/user/src/test_repo/.plzconfig...
23:03:49.081   DEBUG: Read config from /home/user/src/test_repo/.plzconfig
23:03:49.081   DEBUG: Attempting to read config from /home/user/src/test_repo/.plzconfig_linux_amd64...
23:03:49.081   DEBUG: Attempting to read config from /home/user/src/test_repo/.plzconfig.local...
23:03:49.082   DEBUG: Attempting to acquire lock plz-out/.lock...
23:03:49.083   DEBUG: Acquired lock plz-out/.lock
23:03:49.083   DEBUG: Loading built-in build rules...
23:03:49.175   DEBUG: Parser initialised
23:03:49.177   DEBUG: Attempting to read config from .plzconfig_foo_bar...
23:03:49.178   DEBUG: Read config from .plzconfig_foo_bar
23:03:49.180   DEBUG: Registering subrepo pleasings in package //:all
23:03:49.183    INFO: [1/2 tasks] //:foo_bar: Instantiated subrepo [0.0s]
23:03:49.183   DEBUG: Original target scan complete
23:03:49.184   DEBUG: Building ///foo_bar//myproject:_myproject#hdrs...
[...]
```
This is the output with `Arch = foo_bar` in `.plzconfig` but **without** `--arch` on the command line:
```
$ please build -v=4
23:03:30.136   DEBUG: Found repo root at /home/user/src/test_repo
23:03:30.136   DEBUG: Attempting to read config from /etc/please/plzconfig...
23:03:30.136   DEBUG: Attempting to read config from /home/user/.config/please/plzconfig...
23:03:30.136   DEBUG: Attempting to read config from /home/user/src/test_repo/.plzconfig...
23:03:30.136   DEBUG: Read config from /home/user/src/test_repo/.plzconfig
23:03:30.136   DEBUG: Attempting to read config from /home/user/src/test_repo/.plzconfig_linux_amd64...
23:03:30.136   DEBUG: Attempting to read config from /home/user/src/test_repo/.plzconfig.local...
23:03:30.137   DEBUG: Attempting to acquire lock plz-out/.lock...
23:03:30.137   DEBUG: Acquired lock plz-out/.lock
23:03:30.138   DEBUG: Loading built-in build rules...
23:03:30.213   DEBUG: Parser initialised
23:03:30.216   DEBUG: Original target scan complete
23:03:30.218   DEBUG: Need to rebuild //myproject:_myproject#cc, metadata file is missing
23:03:30.235    INFO: //myproject:_myproject#cc: Preparing...
23:03:30.235   DEBUG: Building //myproject:_myproject#hdrs...
23:03:30.236    INFO: //myproject:_myproject#hdrs: Preparing...
23:03:30.236    INFO: [3/7 tasks] //myproject:_myproject#hdrs: Unchanged [0.0s]
23:03:30.270   DEBUG: Registering subrepo pleasings in package //:all
23:03:30.272   DEBUG: Not rebuilding //:_pleasings#download, nothing's changed
23:03:30.272    INFO: //:_pleasings#download: Preparing...
[...]
```

This is the output with `Arch = foo_bar` in `.plzconfig` and `--arch=foo_bar` on the command line:
```
$ please build --arch=foo_bar -v=4
23:03:32.567   DEBUG: Found repo root at /home/user/src/test_repo
23:03:32.568   DEBUG: Attempting to read config from /etc/please/plzconfig...
23:03:32.568   DEBUG: Attempting to read config from /home/user/.config/please/plzconfig...
23:03:32.568   DEBUG: Attempting to read config from /home/user/src/test_repo/.plzconfig...
23:03:32.568   DEBUG: Read config from /home/user/src/test_repo/.plzconfig
23:03:32.568   DEBUG: Attempting to read config from /home/user/src/test_repo/.plzconfig_linux_amd64...
23:03:32.568   DEBUG: Attempting to read config from /home/user/src/test_repo/.plzconfig.local...
23:03:32.570   DEBUG: Attempting to acquire lock plz-out/.lock...
23:03:32.570   DEBUG: Acquired lock plz-out/.lock
23:03:32.570   DEBUG: Loading built-in build rules...
23:03:32.649   DEBUG: Parser initialised
23:03:32.650   DEBUG: Registering subrepo pleasings in package //:all
23:03:32.651    INFO: [1/2 tasks] //:foo_bar: Instantiated subrepo [0.0s]
23:03:32.651   DEBUG: Original target scan complete
23:03:32.651   DEBUG: Building ///foo_bar//myproject:_myproject#hdrs...
23:03:32.651    INFO: ///foo_bar//myproject:_myproject#hdrs: Preparing...
23:03:32.651    INFO: [4/6 tasks] ///foo_bar//myproject:_myproject#hdrs: Unchanged [0.0s]
23:03:32.652   DEBUG: Need to rebuild ///foo_bar//myproject:_myproject#cc, metadata file is missing
23:03:32.655    INFO: ///foo_bar//myproject:_myproject#cc: Preparing...
[...]
```

As you can see the first one is the only one containing these two lines:
```
DEBUG: Attempting to read config from .plzconfig_foo_bar...
DEBUG: Read config from .plzconfig_foo_bar
```
(and the only one that correctly tries to compile using the configuration in `.plzconfig_foo_bar`)

However, in all three cases the debugging output contains:

```
ENVIRONMENT:
[...]
ARCH=bar
GOARCH=bar
GOOS=foo
OS=foo
[...]
XARCH=bar
XOS=foo
[...]
```

Thank you!

**Comment by Tatskaari** [comment]
Thanks for this. Definitely seems like it's doing the wrong thing. Will look into it.

**Comment by Tatskaari** [comment]
Hey, this should be fixed in #1504 

This change is quite risky so I'm probably going to release v15.13.0, then merge that change and release v15.14.0-beta.1
